AzureMonitorAgent/agent.py

```diff
@@ -1087,6 +1149,82 @@ def metrics_watcher(hutil_error, hutil_log):
         finally:
             time.sleep(sleepTime)
 
+def syslogconfig_watcher(hutil_error, hutil_log):
+    """
+    Watcher thread to monitor syslog configuration changes and to take action on them
+    """
+    syslog_enabled  = False
+    # Check for config changes every 30 seconds
+    sleepTime =  30
+
+    # Sleep before starting the monitoring
+    time.sleep(sleepTime)
+
+    while True:
+        try:       
+            if os.path.isfile(AMASyslogConfigMarkerPath):
+                f = open(AMASyslogConfigMarkerPath, "r")
+                data = f.read()
+
+                if (data != ''):
+                    if "true" in data:
+                        syslog_enabled = True
+                f.close()
+
+            if syslog_enabled:
+                # place syslog local configs
+                syslog_enabled  = False
+                hutil_log("generate local syslog configs")
+                generate_localsyslog_configs()
+            else:
+                # remove syslog local configs
+                hutil_log("remove local syslog configs")
+                remove_localsyslog_configs()
+
+        except IOError as e:
+            hutil_error('I/O error in setting up syslog config watcher. Exception={0}'.format(e))
+
+        except Exception as e:
+            hutil_error('Error in setting up syslog config watcher. Exception={0}'.format(e))
+
+        finally:
+            time.sleep(sleepTime)
+
+def generate_localsyslog_configs():
+    """
+    Install local syslog configuration files if not present and restart syslog
+    """
+    if os.path.exists('/etc/rsyslog.d/') and not os.path.exists('/etc/rsyslog.d/10-azuremonitoragent.conf'):
+        copyfile("/etc/opt/microsoft/azuremonitoragent/syslog/05-azuremonitoragent-loadomuxsock.conf","/etc/rsyslog.d/05-azuremonitoragent-loadomuxsock.conf")
+        copyfile("/etc/opt/microsoft/azuremonitoragent/syslog/10-azuremonitoragent.conf","/etc/rsyslog.d/10-azuremonitoragent.conf")
+        os.chmod('/etc/rsyslog.d/05-azuremonitoragent-loadomuxsock.conf', stat.S_IRGRP | stat.S_IRUSR | stat.S_IWUSR | stat.S_IROTH)
+        os.chmod('/etc/rsyslog.d/10-azuremonitoragent.conf', stat.S_IRGRP | stat.S_IRUSR | stat.S_IWUSR | stat.S_IROTH)
+        run_command_and_log(get_service_command("rsyslog", "restart"))
+
+    if os.path.exists('/etc/syslog-ng/syslog-ng.conf') and not os.path.exists('/etc/syslog-ng/conf.d/azuremonitoragent.conf'):
+        syslog_ng_confpath = os.path.join('/etc/syslog-ng/', 'conf.d')
+        os.makedir(syslog_ng_confpath)
+        copyfile("/etc/opt/microsoft/azuremonitoragent/syslog/azuremonitoragent.conf","/etc/syslog-ng/conf.d/azuremonitoragent.conf")
```

Comment:
indent seems off here right?

Reply:
also, do we need to keep syslog-ng config files separately under /etc/opt/microsoft/azuremonitoragent? essentially to avoid cx accidentally manually copying syslog-ng config into rsyslog

Reply:
I ll fix the indent in next iteration. thanks!